mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
Making the change above breaks the `foreachActive` [test](https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala#L281), which counts the explicit zeros as active elements. I can't think of why code would rely on explicit zeros in `foreachActive`, but that's where the potential breakage lies. (The rest of the MLlib tests pass.)